clippy_lints/src/methods/join_absolute_path.rs

```diff
@@ -0,0 +1,30 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, ty::is_type_diagnostic_item};
+use rustc_ast::ast::LitKind;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_span::{symbol::sym::Path, Span};
+
+use super::JOIN_ABSOLUTE_PATH;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>, join_arg: &'tcx Expr<'tcx>, span: Span) {
+    let ty = cx.typeck_results().expr_ty(expr);
+    if_chain!(
+     if is_type_diagnostic_item(cx, ty, Path);
+     let applicability = Applicability::MachineApplicable;
+     if let ExprKind::Lit(spanned) = &join_arg.kind;
+     if let LitKind::Str(symbol, _) = spanned.node;
+     if symbol.as_str().starts_with('/') || symbol.as_str().starts_with('\\');
+     then {
+      span_lint_and_sugg(
+      cx,
+      JOIN_ABSOLUTE_PATH,
+      span.with_hi(expr.span.hi()),
+      r#"argument in join called on path contains a starting '/'"#,
```

Comment:
I would suggest a slightly different message:

```suggestion
      "argument to `Path::join` starts with a path separator",
```

the separator might be platform specific :)